gnoi/gnoi_test.go

```diff
@@ -2997,3 +3002,885 @@ func testGeneratorReflectorCoordination(t *testing.T, linkQualServer *linkQualif
+			stream := &mockGetStream{}
+
+			err := f.Get(tt.request, stream)
+
+			if tt.wantError {
+				if err == nil {
+					t.Errorf("Expected error but got none")
+					return
+				}
+				if status.Code(err) != tt.errorCode {
+					t.Errorf("Expected error code %v, got %v", tt.errorCode, status.Code(err))
+				}
+			} else {
+				if err != nil {
+					t.Errorf("Unexpected error: %v", err)
+				}
+				if len(stream.responses) == 0 {
+					t.Errorf("Expected responses but got none")
+				}
+
+				// Verify content
+				var receivedContent []byte
+				var receivedHash []byte
+				for _, resp := range stream.responses {
+					switch r := resp.Response.(type) {
+					case *fpb.GetResponse_Contents:
+						receivedContent = append(receivedContent, r.Contents...)
+					case *fpb.GetResponse_Hash:
+						receivedHash = r.Hash.Hash
+					}
+				}
+
+				if !bytes.Equal(receivedContent, testContent) {
+					t.Errorf("Content mismatch: expected %s, got %s", testContent, receivedContent)
+				}
+
+				expectedHash := md5.Sum(testContent)
+				if !bytes.Equal(receivedHash, expectedHash[:]) {
+					t.Errorf("Hash mismatch: expected %x, got %x", expectedHash, receivedHash)
+				}
+			}
+		})
+	}
+}
+
+func TestFile_Stat(t *testing.T) {
+	f := newFile()
+
+	// Add test files
+	f.files["/tmp/test1.txt"] = &fileInfo{
+		path:        "/tmp/test1.txt",
+		content:     []byte("content1"),
+		permissions: 0644,
+		created:     time.Now(),
+		modified:    time.Now(),
+	}
+	f.files["/tmp/test2.txt"] = &fileInfo{
+		path:        "/tmp/test2.txt",
+		content:     []byte("content2"),
+		permissions: 0755,
+		created:     time.Now(),
+		modified:    time.Now(),
+	}
+	f.files["/home/user/file.txt"] = &fileInfo{
+		path:        "/home/user/file.txt",
+		content:     []byte("content3"),
+		permissions: 0600,
+		created:     time.Now(),
+		modified:    time.Now(),
+	}
+
+	tests := []struct {
+		name          string
+		request       *fpb.StatRequest
+		wantError     bool
+		errorCode     codes.Code
+		expectedCount int
+	}{
+		{
+			name: "stat /tmp directory",
+			request: &fpb.StatRequest{
+				Path: "/tmp",
+			},
+			wantError:     false,
+			expectedCount: 2,
+		},
+		{
+			name: "stat /home/user directory",
+			request: &fpb.StatRequest{
+				Path: "/home/user",
+			},
+			wantError:     false,
+			expectedCount: 1,
+		},
+		{
+			name: "stat empty directory",
+			request: &fpb.StatRequest{
+				Path: "/empty",
+			},
+			wantError:     false,
+			expectedCount: 0,
+		},
+		{
+			name: "invalid path",
+			request: &fpb.StatRequest{
+				Path: "relative/path",
+			},
+			wantError: true,
+			errorCode: codes.InvalidArgument,
+		},
+		{
+			name: "system path access denied",
+			request: &fpb.StatRequest{
+				Path: "/proc",
+			},
+			wantError: true,
+			errorCode: codes.PermissionDenied,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resp, err := f.Stat(context.Background(), tt.request)
+
+			if tt.wantError {
```

Comment:
normally here we would want to use 

errdiff.Check(err, tt.wantErr)

this saves some of your logic